sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -211,13 +230,21 @@ CURLcode CurlSession::Perform(Context const& context)
     auto hostHeader = headers.find("Host");
     if (hostHeader == headers.end())
     {
-      LogThis("No Host in request headers. Adding it");
+      if (ShouldWrite(LogLevel::Verbose))
+      {
+        Write(LogLevel::Verbose, LogMsgPrefix + "No Host in request headers. Adding it");
+      }
+
       this->m_request.AddHeader("Host", this->m_request.GetUrl().GetHost());
     }
     auto isContentLengthHeaderInRequest = headers.find("content-length");
     if (isContentLengthHeaderInRequest == headers.end())
     {
-      LogThis("No content-length in headers. Adding it");
+      if (ShouldWrite(LogLevel::Verbose))
```

Comment:
You can leave this pattern as is, but since these strings are concatenated at compile time, you could NOT call ShouldLog first and just call Log.